### ISSUE TITLE
llvm-clang-win-x-{armv7l,aarch64}-release: migrate away from DEFAULT_SYSROOT

### ISSUE THREAD

**Issue by MaskRay** [llvm/llvm-project#94284]
The CMake variable `DEFAULT_SYSROOT` provides a default `--sysroot=` for every `clang --target=xxx`.

Within the `clang/test/Driver` testsuite, there have been many changes tha adjust `--sysroot=` so that
these tests will pass on build bots that specify `DEFAULT_SYSROOT`.
Like the now-deprecated `GCC_INSTALL_PREFIX`, this places undue maintenance burden to keep `clang/test/Driver` tests pass for all supported bots.
Two changes should happen:

* There should be no expectation that all tests will pass.
* `DEFAULT_SYSROOT` should eventually be deprecated and then removed. (After we figure out a way to support relative paths https://reviews.llvm.org/D158218)

As alternatives, llvm-clang-win-x-{armv7l,aarch64}-release may switch a Clang configuration file.

* https://clang.llvm.org/docs/UsersManual.html#configuration-files
* https://blogs.gentoo.org/mgorny/2022/10/07/clang-in-gentoo-now-sets-default-runtimes-via-config-file

In short, place `bin/$default_triple.cfg` beside `bin/clang`.

**Comment by vvereschaka** [comment]
Hi @tstellar,

there are few updates for the arm/linux cross cmake cache file -- `clang/cmake/caches/CrossWinToARMLinux.cmake` -- including these that fix the DEFAULT_SYSROOT problem. The post-commit builders work ok with these changes
* https://lab.llvm.org/buildbot/#/builders/119
* https://lab.llvm.org/buildbot/#/builders/60

it would be nice to add these changes into the current release branches to fix the appropriate release bots:
* https://lab.llvm.org/buildbot/#/builders/143
* https://lab.llvm.org/buildbot/#/builders/144

- #87113
- #93363
- #94300
- #94835

https://github.com/llvm/llvm-project/commits/main/clang/cmake/caches/CrossWinToARMLinux.cmake

Please let me know if it requires any specific steps on my side to add these changes into the release branch.

Thanks, Vlad.

**Comment by tstellar** [comment]
@vvereschaka The release/18.x branch is frozen now, we aren't planning any more releases.

**Comment by vvereschaka** [comment]
Understood. Thank you!

**Comment by carlocab** [comment]
> * `DEFAULT_SYSROOT` should eventually be deprecated and then removed. (After we figure out a way to support relative paths https://reviews.llvm.org/D158218)

Relative paths to the configuration file are already supported by something like
```
--sysroot=<CFGDIR>/relative/path/to/sysroot
```

This is, of course, not the same as a driver-relative path, but perhaps it's enough to suit the needs of users who require a relative `DEFAULT_SYSROOT`? Otherwise, I suppose we could support something like `<CLANGDIR>`, but it would probably be nicer if we didn't need it.

**Comment by carlocab** [comment]
wasi-sdk should no longer be using `DEFAULT_SYSROOT` as of WebAssembly/wasi-sdk#503.
